shocker.c

```diff
@@ -1,60 +1,161 @@
 #define _GNU_SOURCE
-#include <sys/wait.h> // waitpid
+#include <stdlib.h>
+#include <stdbool.h>
 #include <sched.h> // clone, all the clone Flags
 #include <unistd.h> // execvp, sethostname
 #include <string.h> // strlen
 #include <stdio.h> // puts, printf
 #include <sys/mount.h> // mount
+#include <sys/stat.h> // mkdir
+#include <sys/types.h> // Flags for sys stuff
+#include <sys/mman.h>
+#include <sys/wait.h> // waitpid
 
-#define STACK_SIZE (1024 * 1024)
+#define STACK_SIZE (1024 * 1024)    /* Stack size for cloned child */
 
 //   docker run image <cmd> <params>
 //./shocker run       <cmd> <params>
 
-static char child_stack[STACK_SIZE]; /* Stack size for cloned child */
-
 int child(void *arg) {
-  sethostname("Container", 10);
-  chdir("/root/container");
-  chroot("/root/container"); // this is the main part of hiding
-  mount("proc", "/proc", "proc", 0, NULL); // proc is a pseudo-fs, used by many tools
+  if (sethostname("Container", 10) == -1) {
```

Comment:
manually providing the length is... not ideal. That being said your only real alternative here is macros:
```
#define my_hostname "Container"
...
sethostname(container_name, sizeof(container_name));
```
or the lengthier and runtime based:
```
static const char *hostname = "blah";
sethostname(hostname, strnlen(hostname, 100));
```
I think I would usually use the later but this is a minor thing here. Usually the value would be dynamic and you'd have to `strnlen` it anyway.